docfx-util.ps1

```diff
@@ -0,0 +1,57 @@
+
+[CmdletBinding(PositionalBinding=$false)]
+param 
+(
+    [parameter(mandatory=$false)][switch][Alias('c')]$clean,
+    [parameter(mandatory=$false)][switch][Alias('d')]$doclinkchecker,
+    [parameter(mandatory=$false)][string][Alias('l')]$lychee,
+    [parameter(mandatory=$false)][string][Alias('a')]$all
+)
+
+
+
+# this is called removeartifacts instead of clean because clean might be taken up by another name
+function removeartifacts
+{
+    $deletePaths = '.\workflows\*.svg', '.\api\*.yml', '.\api\.manifest', '.\_site\', '.\_raw\', '.\_view\'
+    foreach($deletePath in $deletePaths)
+    {    
+        if (Test-Path $deletePath) {
+            Remove-Item $deletePath -Recurse
+        } 
+    }
+}
+
+function doclinkchecker
+{
+    .\build.ps1 --logLevel Warning --warningsAsErrors
+    dotnet DocLinkChecker -v -f .github/workflows/DocLinkChecker.config
+}
+
+function lychee 
+{
+    param($lycheePath)
+    Invoke-Expression "& `"$lycheePath`" --verbose --no-progress --base _site --exclude ^https://github\.com.*merge.* --exclude ^https://github\.com.*apiSpec.* '_site/**/*.html'"
+}
+
+if ($clean)
+{
+    removeartifacts
+}
+
+if ($doclinkchecker)
+{
+    doclinkchecker
+}
+
+If ($PSBoundParameters.ContainsKey('lychee'))
+{
+    lychee($lychee)
+}
+
+If ($PSBoundParameters.ContainsKey('all'))
```

Comment:
See comment above in the README, but this does not run if no path is given:

![image](https://github.com/user-attachments/assets/43016e1d-022f-46b1-9b6b-613333eace9a)

Either update the README to explicitly state the path is required, or allow the `-a` command to run the first two steps without the lychee check links step.

Reply:
I'll update the readme explictly. powershell doesn't seem support calling a script with a `-parameter` without specifying an argument for that parameter. I spent too long looking for an answer on this